### PR TITLE
[FSSDK-9486] maint: Update CI and publishing

### PR DESCRIPTION
**Pull request by mikechu-optimizely**
## Summary
- Large updates to C# CI workflow
- Add release workflow to this repo
- Remove Travis from release build and publish
- Remove AWS S3 for artifact storage and use GitHub's native artifact handling (expires after 90 days, though)

Need to release branch v3.11.3, and I'm taking the opportunity to revamp DevOps

## Test plan

- No code paths changed, so unit & integration tests should continue to work
- Review the recent test [runs](https://github.com/optimizely/csharp-sdk/actions/runs/5487665141) of the workflow

## Issues
- FSSDK-9486

### CODE REVIEW

**Comment by mikechu-optimizely** [comment]
This PR needs to be merged **BEFORE**
#357
#358

File: .github/workflows/csharp.yml [hunk 1]
```diff
@@ -138,14 +96,23 @@ jobs:
...
-        run: |
-          (aws s3 cp ./OptimizelySDK.NetStandard20/bin/Release/netstandard2.0/OptimizelySDK.NetStandard20.dll s3://optly-fs-travisci-artifacts/${{ env.REPO_SLUG }}/${{ env.BUILD_NUMBER }}/${{ env.RUN_NUMBER }}/${{ env.ATTEMPT_NUM }}/OptimizelySDK.NetStandard20.dll-unsigned)
+  
+  integration_tests:
+    name: Run Integration Tests
+    needs: [ netFrameworksAndUnitTest, netStandard16, netStandard20 ]
+    uses: optimizely/csharp-sdk/.github/workflows/integration_test.yml@master
+    secrets:
+      CI_USER_TOKEN: ${{ secrets.CI_USER_TOKEN }}
+      TRAVIS_COM_TOKEN: ${{ secrets.TRAVIS_COM_TOKEN }}
```
Comment (jaeopt) [inline .github/workflows/csharp.yml:108]: Can't we remove all this "TRAVIS" terms? I believe, we do not have any more dependencies onto TRAVIS.

Comment (jaeopt) [reply .github/workflows/csharp.yml:108]: Wondering why we can remove aws s3 for github actions. Can we add why in the PR summary for record?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks great! A couple of suggestions.

**Comment by mikechu-optimizely** [comment]
Later, I'll need to remove the CI and deployment secrets supporting AWS uploads.

**Comment by mikechu-optimizely** [comment]
Temporarily closing to avoid CI runs

File: .github/workflows/csharp.yml (hunk 1 above)
Comment (jaeopt) [reply .github/workflows/csharp.yml:107]: sounds good. we can remove all "TRAVIS" in a separate task.

**Review by jaeopt — APPROVED** [review]
LGTM - so I guess with this change, we now can remove the codesign repo completely?

**Comment by mikechu-optimizely** [comment]
> LGTM - so I guess with this change, we now can remove the codesign repo completely?

@jaeopt I think we should archive it for sure, so it's not used.

**Pull request merged**